book/src/multithreading.md

```diff
@@ -0,0 +1,45 @@
+# Multithreading
+
+RIOT-rs supports multithreading on the Cortex-M, RISC-V, and Xtensa architectures.
+
+## Scheduling
+
+- **Fixed-priority, preemptive scheduling** policy with up to 32 supported priority levels.
+- **Thread priorities are dynamic** and can be changed at runtime.
+- The Runqueue is implemented as a circular linked list for each priority, requiring n × m × 8 bits for _n_ maximum threads and _m_ supported priorities. All operations on the runqueue are performed in constant time, except for the deletion of a thread that is not the head of the runqueue.
+- **Deep sleep when idle**: On single core, no idle threads are created. Instead, if the runqueue is empty, the processor enters deep sleep until a next thread is ready. The context of the previously running thread is only saved once the next thread is ready and the context switch occurs.
```

Comment:
This I'd drop. It is too inaccurate at the moment, we don't "deep leep" yet, just idle the core.
And the lacy context save is an optimization that we should maybe list bundled elsewhere.

Reply:
Agree that _deep sleep_ is wrong, but _sleep mode_ is [correct](https://developer.arm.com/documentation/dui0552/a/the-cortex-m3-processor/power-management/entering-sleep-mode).
I'd argue that it's a relevant difference between single- and multi-core and thus should be mentioned.

But I can drop it if you insist :)

Reply:
ok as is